model/manifest.go

```diff
@@ -288,12 +278,11 @@ func (m *Manifest) GetExecutableForRuntime(goOs, goArch string) string {
 
 	var executable string
 	if server.Executables != nil {
-		if goOs == "linux" && goArch == "amd64" {
-			executable = server.Executables.LinuxAmd64
-		} else if goOs == "darwin" && goArch == "amd64" {
-			executable = server.Executables.DarwinAmd64
-		} else if goOs == "windows" && goArch == "amd64" {
-			executable = server.Executables.WindowsAmd64
+		comb := fmt.Sprintf("%s-%s", goOs, goArch)
+		val, exist := server.Executables[comb]
+
+		if exist {
+			executable = val
```

Comment:
What is the UX for an admin like if no executable was found? Does the error message make that clear?

Reply:
Tracing the code to the supervisor, it looks like the message might be pretty confusing. I'd love to see an explicit call to `GetExecutableForRuntime` inside `newSupervisor` that returns an explicit error when it fails to find an executable for the current architecture.

Reply:
That would mean changing the function return type from `string` to `(string, error)` which also mean breaking semver :/

Reply:
In this case, I think having `GetExecutableForRuntime` in the `model` package was probably unnecessary to begin with. It's not really "shared" code in the sense that we intend anyone other than the server's `plugin` package to execute same. I'd propose copying and implementing a private `getExecutableForRuntime` within the `plugin` package that has the desired return semantics.